### ISSUE TITLE
Provide clearer guidance on how to find an appropriate demo

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#1087]
### Description

Currently, someone looking at our demos lands at https://github.com/stellargraph/stellargraph/tree/develop/demos (or https://github.com/stellargraph/stellargraph/tree/master/demos) and then has to orient themselves and find an appropriate place to visit for their problem. We can try to provide more task-oriented guidance, in addition than just listing the algorithms and high-level task descriptions (which works well for someone familiar with graph ML):

1. someone new to the library and just wants to understand how it works (relates to #992)
2. someone with a specific dataset and problem who may not know all the graph-jargon or the algorithm jargon

For 1, we can provide links to specific notebooks.

For 2, we might provide more generic descriptions of the problems so people can orient themselves. The same applies within each subfolder, e.g. if someone wants to do inductive NAI on a homogeneous graph with node features, how do they work that out from https://github.com/stellargraph/stellargraph/tree/develop/demos/node-classification especially if they're only passingly familiar with what a "homogeneous" graph is in this context.

This relates to #992 and #910.

**Comment by huonw** [comment]
I'm working on this now. My two ideas are:

- a table that has columns like "node classification", "link prediction", "representation learning", "model interpretability", "graph classification" with a row for each algorithm, and then links within the cells as appropriate (this might be better as a rewrite of the top-level README algorithm table)
- a flow-chart/question-answer style list

I'll experiment and see what works nicely.

**Comment by timpitman** [comment]
> * a table that has columns like "node classification", "link prediction", "representation learning", "model interpretability", "graph classification" with a row for each algorithm, and then links within the cells as appropriate (this might be better as a rewrite of the top-level README algorithm table)

I like the idea of a better table if we can make it fit ok in the markdown

**Comment by huonw** [comment]
This is now mostly done with #1200, but there's a few left-over pieces of work in the notes there:

- this only rewrites the basics, node-classification, link-prediction, graph-classification, embeddings, interpretability sub-readmes, the others are left for now to avoid this PR spiralling completely out of control (it would be good to land this before tomorrow morning) and will be done in follow up work
- we should probably add a glossary somewhere, so we don't have to explain/remind what terms like heterogeneous, inductive, ... means
- node-classification/node2vec is a bit weird: it's "really" a unsupervised algorithm and node classification is an downstream task on top of the embeddings computed with it, we could potentially fold it into embeddings (similar to https://github.com/stellargraph/stellargraph/issues/1124)
- community_detection could be folded into embeddings too, because it's also a downstream task on top of unsupervised GraphSAGE (maybe we could have a more explicit "downstream tasks" demo in embeddings, and then all of the others can just refer to that?)

These need to at least be converted into follow up issues before we can call this done.

**Comment by huonw** [comment]
- #1268: Add a glossary to have a central place for definitions/links
- #1269: Community detection demo is a downstream task using embeddings
- #1270: Some demo READMEs don't use the new structure
